File: hangman_js/GuessingStrategy.js
"use strict"

/**
 * A strategy for generating guesses given the current state of a Hangman game.
 */
function GuessingStrategy() {
}

GuessingStrategy.prototype.nextGuess = function(game) {
  throw "Not implemented!";
};

exports.GuessingStrategy = GuessingStrategy;

File: hangman_lua/GuessingStrategy.lua
module(..., package.seeall)

cl = require "cl"

-- A strategy for generating guesses given the current state of a Hangman game.

GuessingStrategy = cl.makeClass {
  nextGuess = function(self, game)
    assert(false, "Not implemented!")
  end
}

return GuessingStrategy
